### PR TITLE
Added rate limit model and handler for password attempt

### PR DESCRIPTION
**Pull request by Scttpr**
Checklist

- [x] I've added a mention in CHANGELOG.md

### CODE REVIEW

File: config.json.dist [hunk 1]
```diff
@@ -36,5 +36,9 @@
   "vapid": {
     "public_key": "",
     "private_key": ""
-  }
+  },
+    "rate_limit": {
+      "attempt_number": 5,
+      "lock_duration": 3600000
```
Comment (Nainterceptor) [inline config.json.dist]: You can reduce this to one hour, not 1000 hours ;p

**Comment by Scttpr** [comment]
[Bruteforce password](https://trello.com/c/mbajA0M8/330-bruteforce-password)

File: helpers/rate-limit-manager.mjs [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    await RateLimit.create({
+      ref, ip,
+    });
+  } else {
+    await userRateLimit.increment(ref, ip);
+  }
+};
+
+export const rateLimitMiddleware = async (ctx, next) => {
+  const userRateLimit = await RateLimit.findOne({ ref: ctx.request.body.email });
```
Comment (Nainterceptor) [inline helpers/rate-limit-manager.mjs]: This code will fail if JSON.parse fail

**Comment by Nainterceptor** [comment]
+ Ensure that IP will be recovered rightly

File: helpers/rate-limit-manager.mjs [hunk 3]
```diff
@@ -0,0 +1,23 @@
+import config from '../services/config';
+import RateLimit from '../models/rate-limit';
+
+export const incrementRateLimit = async (ref, ip) => {
+  const userRateLimit = await RateLimit.findOneAndUpdate({ ref, ip }, { ref, ip }, { upsert: true });
+  if (userRateLimit) {
+    await userRateLimit.increment(ref, ip);
```
Comment (Nainterceptor) [inline helpers/rate-limit-manager.mjs]: Please try to do this increment in the findOneAndUpdate directly.

**Pull request merged**